test/cpp/ext/otel/otel_test_library.h

```diff
@@ -54,31 +54,94 @@ class MockMetricReader : public opentelemetry::sdk::metrics::MetricReader {
   void OnInitialized() noexcept override {}
 };
 
-class OpenTelemetryPluginEnd2EndTest : public ::testing::Test {
- protected:
-  // Note that we can't use SetUp() here since we want to send in parameters.
-  void Init(
-      const absl::flat_hash_set<absl::string_view>& metric_names,
-      opentelemetry::sdk::resource::Resource resource =
-          opentelemetry::sdk::resource::Resource::Create({}),
-      std::unique_ptr<grpc::internal::LabelsInjector> labels_injector = nullptr,
-      bool test_no_meter_provider = false,
-      const std::map<std::string, std::string>& labels_to_inject = {},
-      absl::AnyInvocable<bool(absl::string_view /*target*/) const>
-          target_selector = absl::AnyInvocable<bool(absl::string_view) const>(),
+struct OpenTelemetryPluginTestConfiguration {
+ public:
+  OpenTelemetryPluginTestConfiguration& set_metric_names(
+      absl::flat_hash_set<absl::string_view> names) {
+    metric_names = std::move(names);
+    return *this;
+  }
+
+  OpenTelemetryPluginTestConfiguration& set_resource(
+      const opentelemetry::sdk::resource::Resource& res) {
+    resource = res;
+    return *this;
+  }
+
+  OpenTelemetryPluginTestConfiguration& set_labels_injector(
+      std::unique_ptr<grpc::internal::LabelsInjector> injector) {
+    labels_injector = std::move(injector);
+    return *this;
+  }
+
+  OpenTelemetryPluginTestConfiguration& set_test_no_meter_provider(bool flag) {
```

Comment:
I suggest inverting this: call the parameter `use_meter_provider`, make it true by default, and use this method to disable it when appropriate.

Reply:
note that there is just one test that needs to disable meter provider. All others need it, which means that all other tests would need to set this option if we invert it.

Reply:
I'm not suggesting changing the default, just inverting the semantics to avoid a double negative.

Right now, the one test that needs to not use a MeterProvider does `set_test_no_meter_provider(true)`.  What I'm suggesting is that that test should instead say `set_use_meter_provider(false)`.  No other test should need any changes.

Reply:
ah, misread the comment

Reply:
done